includes/fraud-prevention/class-fraud-prevention-service.php

```diff
@@ -88,7 +88,7 @@ public static function maybe_append_fraud_prevention_token() {
 
 		// Don't add the token if the user isn't on the cart or checkout page.
 		// Checking the cart page too because the user can pay quickly via the payment buttons on that page.
-		if ( ! is_checkout() && ! is_cart() ) {
+		if ( ! is_checkout() && ! is_cart() && ! is_product() ) {
```

Comment:
Nice catch! Would be nice to add a comment here providing context as to why we're skipping the fraud prevention token in the product page as well.

Reply:
We are not skipping it. Here we are adding the Fraud Prevention Token to the product page to make it work. Without it, the order wouldn’t process. This change addresses issue #8981.

Reply:
Sorry, I misread the comment. But what I mean by that is that we should add a comment mentioning why we're adding the fraud prevention token to the product page as well (similar to the cart). Something like:

```diff
+ // Checking the product page too because the user can pay through express checkout (ECE/PRB) on that page.
```

This is just to provide context to folks who are not aware of why the token is needed in there.

Reply:
Note that this token is not exclusive to ECE or PRBs. While express checkouts use it, it is intended for use on any page that submits an order for processing. Done in https://github.com/Automattic/woocommerce-payments/pull/8987/commits/bcf8bfa43c1e827836b15de524aeb24c05db3f71